packages/subiquity_client/lib/src/client.dart

```diff
@@ -577,6 +577,21 @@ class SubiquityClient {
     return _receive('getActiveDirectoryJoinResult($wait)', request,
         AdJoinResult.values.byName);
   }
+
+  Future<void> restart() async {
+    final request = await _openUrl('POST', 'meta/restart');
+    try {
+      await request.close();
+    } on HttpException catch (e) {
+      if (e.message
+          .contains('Connection closed before full header was received')) {
+        _log.info(
+            'Subiquity closed the connection before the response was received. This is expected.');
```

Comment:
What a strange behavior

Reply:
Not sure if there's something weird in dart's http client, I didn't investigate it further

Reply:
I bet the weird behavior is on subiquity's side, that they close down everything before they got the ack